File: it.unibo.java.radar/src/it/unibo/radar/common/radarSupport.java
package it.unibo.radar.common;
import it.unibo.is.interfaces.IOutputEnvView;
 
 
public class radarSupport {
private  static RadarControl radarControl;
	
 
	public static void setUpRadarGui( ) {
		try {
			radarControl = new RadarControl( null );
		} catch (Exception e) {
 			e.printStackTrace();
		}
	}
	public static void update(  String dist, String theta ){
		if( radarControl != null ) radarControl.update(   dist,   theta );
	}
//	public static void sendDataToGui(  String distance, String angle ){
// 		update(   qa, ""+distance,   ""+angle );
//	}
}

File: it.unibo.eclipse.qak.robot/src/resources/radarSupport.kt
package resources

import it.unibo.radar.common.radarSupport.*

object radarSupport{
		 fun activate(){
			println("ACTIVATE THE RADAR")
			setUpRadarGui()
 		}		
		fun spot( distance : Int, angle: Int ){
			update(distance.toString(),angle.toString())
		}
		fun spot( distance : String, angle: String ){
			update(distance,angle)
		}

	/*
	lateinit var radar : RadarControl 
	
	fun create(){
		it.unibo.radar.common.radarSupport.setUpRadarGui()
		radar = RadarControl(null)
	}
	fun sendData( dist: String, theta: String ){
		val d = Integer.parseInt(dist)*5	//Amplify
		radar.update(   "$d",   theta );
	}
 */
}